### ISSUE TITLE
@mock_cloudformation doesn't work

### ISSUE THREAD

**Issue by jhw** [spulec/moto#5462]
### requirements.txt

```
moto
pyyaml
openapi-spec-validator
```

### hello.py

```
from moto import mock_cloudformation

import unittest

@mock_cloudformation
class HelloTest(unittest.TestCase):

    def test_hello(self):
        self.assertEqual(2, 2)
    
if __name__=="__main__":
    unittest.main()
```

### run

```
(env) me@mine % python hello.py                             
Traceback (most recent call last):
  File "hello.py", line 6, in <module>
    class HelloTest(unittest.TestCase):
  File "/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/moto/__init__.py", line 8, in f
    module = importlib.import_module(module_name, "moto")
  File "/Users/me/.pyenv/versions/3.8.11/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/moto/cloudformation/__init__.py", line 1, in <module>
    from .models import cloudformation_backends
  File "/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/moto/cloudformation/models.py", line 18, in <module>
    from .parsing import ResourceMap, OutputMap
  File "/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/moto/cloudformation/parsing.py", line 17, in <module>
    from moto.apigateway import models  # noqa  # pylint: disable=all
  File "/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/moto/apigateway/__init__.py", line 1, in <module>
    from .models import apigateway_backends
  File "/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/moto/apigateway/models.py", line 15, in <module>
    from openapi_spec_validator.exceptions import OpenAPIValidationError
ImportError: cannot import name 'OpenAPIValidationError' from 'openapi_spec_validator.exceptions' (/Users/me/work/gists/aws/moto_cloudformation/env/lib/python3.8/site-packages/openapi_spec_validator/exceptions.py)
```

**Comment by bblommers** [comment]
Use `moto[cloudformation]` to install dependencies.
See http://docs.getmoto.org/en/latest/docs/getting_started.html#installing-moto

**Comment by yan12125** [comment]
From the error message, this seems the same issue as https://github.com/spulec/moto/issues/5457, which was recently fixed, but not included in the current version yet.

**Comment by bblommers** [comment]
Ah, you are right @yan12125, I misread the error.

@jhw This is fixed in moto >= 4.0.3.dev6 - I'll try to get a full release out today.

**Comment by bblommers** [comment]
Moto 4.0.3 is now available on PyPi! I'll close this, but please let us know if you run into any other issues.

**Comment by jhw** [comment]
This works now, many thanks :)

Couple of things I noticed in trying to deploy a random template under `@mock_cloudformation` -

- `AWS::ApiGateway::Method` seems to require the `ApiKeyRequired` field; this is listed as `Required: No` in the AWS docs -

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-method.html#cfn-apigateway-method-apikeyrequired

- `AWS::ApiGateway::Deployment` seems to require the `StageName` field; this is also listed as `Required: No` in the AWS docs - 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-deployment.html#cfn-apigateway-deployment-stagename

Many thanks for the `@mock_cloudformation` fix :)

**Comment by bblommers** [comment]
Thanks for raising them @jhw! All three issues should be fixed as of moto >= 4.0.4.dev2
